test/e2e/main_test.go

```diff
@@ -465,7 +468,73 @@ func testServerTLS(ctx context.Context, namespace string) func(t *testing.T) {
 	}
 }
 
-// TestIsManagedByController test prometheus operator managing object with correct ControlerID.
+// testOperatorMetrics checks whether prometheus operator metrics exist after all the end-to-end tests.
+func testOperatorMetrics(ctx context.Context, namespace string) func(t *testing.T) {
+	return func(t *testing.T) {
+		skipPrometheusAllNSTests(t)
+
+		if err := framework.WaitForServiceReady(context.Background(), namespace, prometheusOperatorServiceName); err != nil {
+			t.Fatal("waiting for prometheus operator service: ", err)
+		}
+
+		operatorService := framework.KubeClient.CoreV1().Services(namespace)
+		request := operatorService.ProxyGet("https", prometheusOperatorServiceName, "https", "/metrics", make(map[string]string))
+		resp, err := request.DoRaw(ctx)
+		if err != nil {
```

Comment:
Can you use `require` package here rather than using `t.Fatal` statements? 
https://pkg.go.dev/github.com/stretchr/testify/require

 This approach is way better and also reduces the number of lines.

Reply:
Thanks for the review and guidance! I will change it to that 😄

Reply:
resolved in my latest commit 😄